mdal/frmts/mdal_hec2d.cpp

```diff
@@ -761,18 +760,30 @@ std::unique_ptr<MDAL::Mesh> MDAL::DriverHec2D::load( const std::string &resultsF
   {
     HdfFile hdfFile = openHdfFile( mFileName );
 
-    // Verify it is correct file
-    std::string fileType = openHdfAttribute( hdfFile, "File Type" );
-    bool oldFormat = canReadOldFormat( fileType );
-
     HdfGroup gGeom = openHdfGroup( hdfFile, "Geometry" );
     HdfGroup gGeom2DFlowAreas = openHdfGroup( gGeom, "2D Flow Areas" );
 
     std::vector<std::string> flowAreaNames;
-    if ( oldFormat )
-      flowAreaNames = read2DFlowAreasNamesOld( gGeom2DFlowAreas );
-    else
-      flowAreaNames = read2DFlowAreasNames505( gGeom2DFlowAreas );
+
+    try
+    {
+      flowAreaNames = read2DFlowAreasNamesFromNameDataset( gGeom2DFlowAreas );
```

Comment:
can we change this read2DFlowAreasNamesFromNameDataset() and read2DFlowAreasNamesFromAttributesDataset() to just return empty flowAreaNames on error and not throw? 

it is generally not advisable to use try-catch for expected code-flow and not error situation.

Reply:
done